### ISSUE TITLE
Keras ImageDataGenerator doesn't work properly with shuffle when transforming image and mask together

### ISSUE THREAD

**Issue by AlbertoSabater** [keras-team/keras#7702]
I have created two ImageDataGenerator objects to process two images at the same time (one image and his mask) to train an autoencoder/u-net. To randomize the dataset I set `shuffle=True`. I take these images from two folders (one for the image and other for the mask), so I use two `flow_from_directory` with the same seed. However, in practice that doesn't work and the mask extracted doesn't correspond to the extracted image. Both images (image and mask) have the same file name.

The problem is that the filenames from the data directories are read in a different order. To solve that I have added the next line to the DirectoryIterator at the end of his __init__ function.
`        self.filenames.sort()
`

The problem seems to be solved.

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).
